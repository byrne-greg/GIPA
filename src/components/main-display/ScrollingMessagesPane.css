.ScrollingMessagesPane {
	overflow: auto;
	min-height: 200px;
	max-height: 400px;
	margin: 10px;
}

.ScrollingMessagesPane__botMsg, .ScrollingMessagesPane__userMsg {
	position: relative;
	border-radius: .4em;
	padding: 20px;
	color: white;
	min-width: 5rem;
	margin: 20px 2rem;
	clear:both;
	text-align: left;
}

.ScrollingMessagesPane__botMsg p, .ScrollingMessagesPane__userMsg p {
	margin: 5px;
}


.ScrollingMessagesPane__botMsg {
	background: #007fa5;
	float:left;
}

.ScrollingMessagesPane__userMsg {
	background: #73348E;
	float:right;	
}

.ScrollingMessagesPane__botMsg:after, .ScrollingMessagesPane__userMsg:after {
	content: '';
	position: absolute;
	right: 0;
	top: 50%;
	width: 0;
	height: 0;
	border: 25px solid transparent;	
}

.ScrollingMessagesPane__userMsg:after{
	right:0;
	border-left-color: #73348E;
	border-right: 0;
	margin-top: -25px;
  	margin-right: -25px;
}

.ScrollingMessagesPane__botMsg:after {
	left: 0;
	border: 25px solid transparent;
	border-right-color: #007fa5;
	border-left: 0;
	margin-top: -25px;
	margin-left: -25px
}