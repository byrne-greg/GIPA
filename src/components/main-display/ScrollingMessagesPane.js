import "./ScrollingMessagesPane.css";
import React from "react";
import PropTypes from "prop-types";
import shortid from "shortid";

export default class ScrollingMessagesPane extends React.Component {
 
	componentDidUpdate() {
		this.scrollToBottom();
	}

	componentDidMount() {
		this.scrollToBottom();
	}

	scrollToBottom() {
		if(document.getElementById("latest-message") !== null) {
			document.getElementById("latest-message").scrollIntoView({ behavior: "smooth" });
		}
	}
    
	createMessage(msgObj) {
		let msgType = (msgObj.type === "user" ? "userMsg" : "botMsg");
		return (
			<Message key={shortid.generate()} text={msgObj.message} type={msgType}/>
		);     
	}

	displayMessages(messages) {
		if(messages && messages !== null && messages !== undefined) {
			return(
				messages.map(m => (this.createMessage(m))));
		} else {
			return( <h2>No Messages to Display</h2>);
		}
      
	}
    
	render() {
		return(
			<div className="ScrollingMessagesPane">
				{/* {this.props.chatMessages.map(m => (this.createMessage(m)))} */}
				{/* dummy div used to "scroll Into View" */}
				{this.displayMessages(this.props.chatMessages)}
				<div id="latest-message" style={{ float:"left", clear: "both" }}/> 
			</div>
		);
	}

}

export const Message = props => {
	return (
		<div className={"ScrollingMessagesPane__" + props.type}>
			<p>{props.text}</p>
		</div>
	);
};
Message.propTypes = {
	type: PropTypes.string.isRequired,
	text: PropTypes.string.isRequired
};

 