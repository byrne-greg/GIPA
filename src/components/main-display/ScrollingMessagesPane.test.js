import React from "react";
import ScrollingMessagesPane, {Message} from "./ScrollingMessagesPane.js";
import renderer from "react-test-renderer";
import {shallow} from "enzyme/build";

describe("ScrollingMessagesPane component", () => {

	it("shallow renders without crashing", () => {
		shallow(<ScrollingMessagesPane/>);
	});

	it("full render matches expected snapshot", () => {
		const component = renderer.create(<ScrollingMessagesPane/>).toJSON();
		expect(component).toMatchSnapshot();
	});

});

describe("Message component", () => {

	it("shallow renders without crashing", () => {
		shallow(<Message text="test" type="test"/>);
	});

	it("shallow renders with given css type", () => {
		let cssMessageType = "cssMessageType";
		var wrapper = shallow(<Message text="test" type={cssMessageType}/>);
		expect(wrapper.find("div").prop("className")).toBe("ScrollingMessagesPane__" + cssMessageType);
	});

	it("full render matches expected snapshot", () => {
		const component = renderer.create(<Message text="test" type="cssMessageType"/>).toJSON();
		expect(component).toMatchSnapshot();
	});

});