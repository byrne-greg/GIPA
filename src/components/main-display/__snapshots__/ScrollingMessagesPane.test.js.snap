// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component full render matches expected snapshot 1`] = `
<div
  className="ScrollingMessagesPane__cssMessageType"
>
  <p>
    test
  </p>
</div>
`;

exports[`ScrollingMessagesPane component full render matches expected snapshot 1`] = `
<div
  className="ScrollingMessagesPane"
>
  <h2>
    No Messages to Display
  </h2>
  <div
    id="latest-message"
    style={
      Object {
        "clear": "both",
        "float": "left",
      }
    }
  />
</div>
`;
